yang/frr-zebra.yang

```diff
@@ -0,0 +1,1887 @@
+module frr-zebra {
+  yang-version 1.1;
+  namespace "http://frrouting.org/yang/zebra";
+  prefix frr-zebra;
+
+  import ietf-yang-types {
+    prefix yang;
+  }
+
+  import ietf-inet-types {
+    prefix inet;
+  }
+
+  import frr-route-types {
+    prefix frr-route-types;
+  }
+
+  import ietf-routing-types {
+    prefix rt-types;
+  }
+
+  organization
+    "Free Range Routing";
+  contact
+    "FRR Users List: <mailto:frog@lists.frrouting.org>
+     FRR Development List: <mailto:dev@lists.frrouting.org>";
+  description
+    "This module defines a model for managing the FRR zebra daemon.";
+
+  revision 2019-06-01 {
+    description
+      "Initial revision.";
+    reference
+      "ISO/IEC 10589:2002.";
+  }
+
+  typedef unix-timestamp {
+    type uint32;
+    units "seconds";
+    description
+      "An absolute time in seconds since the unix epoch.";
+  }
+
+  /*
+   * Multicast RPF mode configurable type
+   */
+  typedef mcast-rpf-lookup-mode {
+    type enumeration {
+      enum "none" {
+	value 0;
+	description
+	  "No mode set.";
+      }
+      enum "mrib-only" {
+	value 1;
+	description
+	  "Lookup in unicast RIB only.";
+      }
+      enum "urib-only" {
+	value 2;
+	description
+	  "Lookup in multicast RIB only.";
+      }
+      enum "mrib-then-urib" {
+	value 3;
+	description
+	  "Try multicast RIB first, fall back to unicast RIB.";
+      }
+      enum "lower-distance" {
+	value 4;
+	description
+	  "Lookup both unicast and mcast, use entry with lower distance.";
+      }
+      enum "longer-prefix" {
+	value 5;
+	description
+	  "Lookup both unicast and mcast, use entry with longer prefix.";
+      }
+    }
+    description
+      "Multicast RPF lookup behavior";
+  }
+
+  typedef nexthop-type {
```

Comment:
Numerous other protocols use nexthops.  Perhaps this is a logical break out point?  As that this won't be specific to zebra?

Reply:
It does seem like all the protocols should have a common nexthop type they are using here -- it would make for a more consistent interface for scripts, etc., in the long run, and prevent "name drift" among different protocols.

Reply:
Yes, that makes good sense - I'll move the nexthop info to a shareable lib module.